database_impl.go

```diff
@@ -158,3 +159,38 @@ func (d *database) ValidateQuery(ctx context.Context, query string) error {
 	}
 	return nil
 }
+
+func (d *database) Transaction(ctx context.Context, action string, options *TransactionOptions) (interface{}, error) {
+	req, err := d.conn.NewRequest("POST", path.Join(d.relPath(), "_api/transaction"))
+	if err != nil {
+		return nil, WithStack(err)
+	}
+	input := transactionRequest{Action: action}
+	if options != nil {
+		input.MaxTransactionSize = options.MaxTransactionSize
+		input.LockTimeout = options.LockTimeout
+		input.WaitForSync = options.WaitForSync
+		input.IntermediateCommitCount = options.IntermediateCommitCount
+		input.Params = options.Params
+		input.IntermediateCommitSize = options.IntermediateCommitSize
+		input.Collections.Read = options.ReadCollections
+		input.Collections.Write = options.WriteCollections
+	}
+	if _, err = req.SetBody(input); err != nil {
```

Comment:
When using an `if` with a nested variable lik this, I prefer to have the nested variable defined explicitly.
Like:

`if _, err := req.SetBody(input); err != nil {...`

Feel free to change. This is not a show stopper.